### PR TITLE
Created parser and classes with SingleTon class

### PR DESCRIPTION
**Pull request by mich-chmielewski**
Version of parser with read from file, read from url and write to local file

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: pom.xml [hunk 1]
```diff
@@ -24,6 +24,11 @@
             <artifactId>log4j-slf4j-impl</artifactId>
             <version>2.12.1</version>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.9.8</version>
```
Comment (rafalmisiak) [inline pom.xml]: 2.10.1 jest nowszą wersją

File: src/main/java/com/infoshare/academy/highfive/HolidayDate.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class HolidayDate {
+    @JsonAlias({ "date", "iso" })
+   @JsonIgnoreProperties(value = {"date"})
```
Comment (rafalmisiak) [inline src/main/java/com/infoshare/academy/highfive/HolidayDate.java]: formatowanie

File: src/main/java/com/infoshare/academy/highfive/Holiday.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public void setDate(HolidayDate date) {
+        this.date = date;
+    }
+
+    public List<String> getType() {
+        return type;
+    }
+
+    public boolean isfree() {
+        return type.contains("National holiday");
```
Comment (rafalmisiak) [inline src/main/java/com/infoshare/academy/highfive/Holiday.java]: isFree

File: src/main/java/com/infoshare/academy/highfive/Holiday.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package com.infoshare.academy.highfive;
+
+import java.util.List;
+public class Holiday {
+
+    private String name;
+    private String description;
+    private HolidayDate date;
+    private List<String> type = null;
```
Comment (rafalmisiak) [inline src/main/java/com/infoshare/academy/highfive/Holiday.java]: Typ do enuma

**Pull request closed without merging**